test/cppsuite/test_harness/core/configuration.h

```diff
@@ -44,8 +44,10 @@ class configuration {
     public:
     configuration(const std::string &test_config_name, const std::string &config)
     {
-        std::string default_config =
-          std::string(__wt_test_config_match(test_config_name.c_str())->base);
+        const auto *config_entry = __wt_test_config_match(test_config_name.c_str());
+        if (config_entry == nullptr)
+            testutil_die(EINVAL, "failed to match test config name");
```

Comment:
@luke-pearson, we could use `testutil_assert(cfg_entry != nullptr);` instead of the if/testutil_die(), what do you think ?

Reply:
I think die is better in this case as it prints a meaningful message. If we just assert could be confusing?

Reply:
True - we have a few testutil_assert in the code that we may want to change at some point then.